src/control/BarrettFingerKinematicSimulationSpreadCommandExecutor.cpp

```diff
@@ -159,18 +156,32 @@ void BarrettFingerSpreadCommandExecutor::step(double _timeSincePreviousCall)
     mInExecution = false;
     return;
   }
-  
+
   // Move spread
   double newSpread;
   if (spread > mGoalPosition)
-    newSpread = std::max(mGoalPosition, spread + -kDofVelocity*_timeSincePreviousCall);
+    newSpread = std::max(mGoalPosition,
+      spread + -kDofSpeed*period);
   else
-    newSpread = std::min(mGoalPosition, spread + kDofVelocity*_timeSincePreviousCall);
-  
+    newSpread = std::min(mGoalPosition,
+      spread + kDofSpeed*period);
+
   for (auto spreadDof: mSpreadDofs)
     spreadDof->setPosition(newSpread);
-
 }
 
+//=============================================================================
+bool BarrettFingerKinematicSimulationSpreadCommandExecutor::setCollideWith(
+  ::dart::collision::CollisionGroupPtr collideWith)
+{
+  std::lock_guard<std::mutex> lockSpin(mMutex);
+
```

Comment:
Verify (or make) `collideWIth->getCollisionDetector()` be consistent with `mCollisionDetector`.